dataset_reader/base_reader.py

```diff
@@ -1,21 +1,45 @@
 from dataclasses import dataclass
 from typing import Iterator, List, Optional
 
+from scipy.sparse import spmatrix as SparseMatrix
```

Comment:
would it be possible to avoid `scipy`? having just python dict / couple of list should be ore convenient

Reply:
I tried but looks like I'll have to re-write the same features as `csr_matrix()` to be able to parse out `indices` and `values` for each sparse vector. I'd rather introduce and return a `SparseVector` to keep it clean.